Add more resource info to Pod details

joaquimrocha opened the pull request:
- frontend: Add link to node in Pod details view
- frontend: Use StatusLabel component for the Pod status in details view

joaquimrocha commented:
I have updated this PR with a new entry in the metadata display table that shows owner refs/controlled by labels.

illume commented on docs/platforms.md:
```diff
@@ -0,0 +1,20 @@
+# Tested Platforms
```
Was this doc added accidentally?

joaquimrocha replied:
Yes. I noticed the mistake and added it to the right branch: https://github.com/kinvolk/headlamp/pull/256
But forgot to force push this one. Updated now.

illume commented on frontend/src/components/common/Resource/Resource.tsx:
```diff
@@ -58,6 +60,33 @@ interface MetadataDisplayProps {
 export function MetadataDisplay(props: MetadataDisplayProps) {
   const { resource, extraRows } = props;
 
+  function makeOwnerReferences(ownerReferences: KubeOwnerReference[]) {
```
Could drop the `make` part here - since it's basically a component it should be Capitalized.

illume commented on frontend/src/lib/k8s/cluster.ts:
```diff
@@ -37,6 +37,16 @@ export interface KubeMetadata {
+  ownerReferences?: KubeOwnerReference[];
+}
+
+export interface KubeOwnerReference {
```
Documenting the interface fields would be helpful. 

```typescript
  /** */
  apiVersion: string;
```

joaquimrocha replied:
This is basically a copy from the [K8s reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.21/). I don't think there's a lot of value by adding the docs. Maybe adding a link + context in a comment at the top should be enough.

illume reviewed (changes requested):
Could you please add a storybook example to show the change?

joaquimrocha commented:
@illume , I have now added a story for the resource's MetadataDisplay component. Please check it out.

illume commented:
In 7a1bd26ebefa2e6d6858584eddaa0b5ca2e1976a I added config item so that timezone and locale are set to a default en-us/UTC when `npm run test`. So the tests pass locally, and on CI for me. I'd suggest this change would be good to apply to this PR so the tests pass.

ps. Maybe it can be fixed in a more elegant way in the i18n branch, where this date formatting function usage will be replaced with formatting through the i18next way. There's a way to also mock things with jest (and in combination with storybook as in here: https://github.com/storybookjs/storybook/issues/3809#issuecomment-592759605) as well. I'll be pushing these to the i18n branch.

pps. Perhaps we should display the timezone when showing dates? Since it's perhaps semi-confusing thing knowing which timezone a date is in (UTC, some other random server time, or browser time). I know for me, setting servers to UTC I sometimes assume the dates will be also displayed in UTC. I know deciding how date formatting should be done is probably out of scope of this PR... just thought I'd mention it.

joaquimrocha commented:
> In [7a1bd26](https://github.com/kinvolk/headlamp/commit/7a1bd26ebefa2e6d6858584eddaa0b5ca2e1976a) I added config item so that timezone and locale are set to a default en-us/UTC when `npm run test`. So the tests pass locally, and on CI for me. I'd suggest this change would be good to apply to this PR so the tests pass.

Cool. Hadn't noticed that.
 
> ps. Maybe it can be fixed in a more elegant way in the i18n branch, where this date formatting function usage will be replaced with formatting through the i18next way. There's a way to also mock things with jest (and in combination with storybook as in here: [storybookjs/storybook#3809 (comment)](https://github.com/storybookjs/storybook/issues/3809#issuecomment-592759605)) as well. I'll be pushing these to the i18n branch.

Let's do that later.

> pps. Perhaps we should display the timezone when showing dates? Since it's perhaps semi-confusing thing knowing which timezone a date is in (UTC, some other random server time, or browser time). I know for me, setting servers to UTC I sometimes assume the dates will be also displayed in UTC. I know deciding how date formatting should be done is probably out of scope of this PR... just thought I'd mention it.

For showing more details, I think we should do that on hover, either on the date itself or on a watch icon next to it. But yeah, I agree having a quick way to know the TZ is important when dealing with services.

joaquimrocha commented:
@illume , I squashed your commit with mine so we don't have a broken tests commit that we could avoid. I added you as co-author in the commit. Thanks.
